tests/run-pass/ptr_int_roundtrip.rs

```diff
@@ -0,0 +1,32 @@
+// compile-flags: -Zmiri-permissive-provenance -Zmiri-disable-stacked-borrows
+#![feature(strict_provenance)]
+
+fn ptr_roundtrip_out_of_bounds() {
+    let x: i32 = 3;
+    let x_ptr = &x as *const i32;
+
+    let x_usize = x_ptr.wrapping_offset(128) as usize;
```

Comment:
Please write this test entirely with `expose_addr` and `from_exposed_addr` (rather than `as` casts). I think that should make it more clear what happens.